lib/HLSL/DxilOperations.cpp

```diff
@@ -718,6 +727,15 @@ void OP::RemoveFunction(Function *F) {
   }
 }
 
+OP::OpCodeClass OP::GetOpClassForDxilFunction(const Function *F) {
+  auto iter = m_FunctionToOpClass.find(F);
+  if (iter == m_FunctionToOpClass.end()) {
```

Comment:
If we just keep running we will dereference out-of-bounds memory with `iter->second` (since `iter == end()`. Using an if lets us return an known-invalid opcode class and perhaps prevent a crash in fre.